pkg/registry/controller/etcd/etcd.go

```diff
@@ -37,7 +38,7 @@ func NewREST(s storage.Interface, storageFactory storage.StorageFactory) (*REST,
 
 	newListFunc := func() runtime.Object { return &api.ReplicationControllerList{} }
 	storageInterface := storageFactory(
-		s, 100, nil, &api.ReplicationController{}, prefix, true, newListFunc)
+		s, 100, etcdstorage.APIObjectVersioner{}, &api.ReplicationController{}, prefix, true, newListFunc)
```

Comment:
We need to pass Versioner to the cacher (it's necessary to serve List from the cache).
There are few options:
1 the one above (but this will introducing etcd dependency in all pkg/registry/foo directories)
2 move "storage.StorageFactory" to somewhere else, and pass it all resource already filled with APIObjectVersioner (this will introduce dependency only in pkg/master/master.go where we will be building storageFactory object)
3 I though about something else but fogot :(

Thoughts on it?